src/de.rs

```diff
@@ -0,0 +1,492 @@
+use std::collections::VecDeque;
+use std::ops::{AddAssign, MulAssign, Neg};
+
+use serde::de::{self, Deserialize, DeserializeSeed, SeqAccess, Visitor};
+
+use rlp::{self, ExpectedType};
+
+use error::{Error, Result};
+use std::str;
+
+pub struct Deserializer<'de> {
+    input: &'de [u8],
+    /// Stacked input slices for nested data
+    stack: VecDeque<&'de [u8]>,
+}
+
+impl<'de> Deserializer<'de> {
+    pub fn from_bytes(input: &'de [u8]) -> Self {
+        Deserializer {
+            input: input,
+            stack: VecDeque::new(),
+        }
+    }
+}
+
+pub fn from_bytes<'a, T>(s: &'a [u8]) -> Result<T>
+where
+    T: Deserialize<'a>,
+{
+    let mut deserializer = Deserializer::from_bytes(s);
+    let t = T::deserialize(&mut deserializer)?;
+    if deserializer.input.len() == 0 {
+        Ok(t)
+    } else {
+        Err(Error::TrailingBytes)
+    }
+}
+
+impl<'de> Deserializer<'de> {
+    // Parse the JSON identifier `true` or `false`.
+    fn parse_bool(&mut self) -> Result<bool> {
+        unimplemented!();
+    }
+
+    fn parse_unsigned<T>(&mut self) -> Result<T>
+    where
+        T: AddAssign<T> + MulAssign<T> + From<u8>,
+    {
+        unimplemented!();
+    }
+
+    fn parse_signed<T>(&mut self) -> Result<T>
+    where
+        T: Neg<Output = T> + AddAssign<T> + MulAssign<T> + From<i8>,
+    {
+        // Optional minus sign, delegate to `parse_unsigned`, negate if negative.
+        unimplemented!()
+    }
+
+    fn parse_string(&mut self) -> Result<&'de str> {
+        let res = rlp::decode_length(&self.input)?;
+        if res.expected_type == ExpectedType::StringType {
+            let s = str::from_utf8(&self.input[res.offset..res.offset + res.length])
+                .map_err(|_| Error::InvalidString)?;
+            self.input = &self.input[res.offset + res.length..];
+            Ok(s)
+        } else {
+            Err(Error::ExpectedString)
+        }
+    }
+
+    fn parse_bytes(&mut self) -> Result<&'de [u8]> {
+        let res = rlp::decode_length(&self.input)?;
+        if res.expected_type == ExpectedType::StringType {
+            // let s = str::from_utf8(&self.input[res.offset..res.offset + res.length])
```

Comment:
why is this left commented out here?

Reply:
Going to remove this one. A lot of trial and error I guess.